src/services/inlayHints.ts

```diff
@@ -3,6 +3,10 @@ namespace ts.InlayHints {
 
     const maxHintsLength = 30;
 
+    const leadingParameterNameCommentRegexFactory = (name: string) => {
+        return new RegExp(`^\\s?/\\*\\*?\\s?${name}\\s?\\*\\/\\s?$`);

```

Comment:
Do we need to sanitize this name before plugging it into a regular expression? There are times when we put non-identifier text into an identifier to help with parser recovery, e.g. leading `#` when you try to use a private identifier where one can’t actually be parsed. I can’t think of a scenario where we put any RegExp control characters in, but maybe it could happen in the future. It would probably be fine just to not run this check if `name` is not valid identifier text.

Reply:
I guess check the name is a identifierText  would cover this.